spacelift/resource_worker_pool.go

```diff
@@ -0,0 +1,129 @@
+package spacelift
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/structs"
+)
+
+func resourceWorkerPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWorkerPoolCreate,
+		Read:   resourceWorkerPoolRead,
+		Update: resourceWorkerPoolUpdate,
+		Delete: resourceWorkerPoolDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"config": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "credentials necessary to connect WorkerPool's workers to the control plane",
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "name of the worker pool",
+				Required:    true,
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "description of the worker pool",
+				Optional:    true,
+				Default:     "",
```

Comment:
I don't think you actually want to set a default value to an empty string. Instead, I'd suggest checking for the field value being set.